test/integration/spec/connect.js

```diff
@@ -104,41 +104,53 @@ describe('connect', function() {
     });
   });
 
-  describe('region selection', async () => {
-    ['without', 'global', 'specific', 'invalid'].forEach(regionType => {
+  describe('signaling region', async () => {
+    const regions = ['invalid', 'without', 'gll'].concat(defaults.regions ? defaults.regions.split(',') : [
+      'au1', 'br1', 'de1', 'ie1', 'in1', 'jp1', 'sg1', 'us1', 'us2'
+    ]);
+
+    regions.forEach(region => {
       const regionOptions = {
-        global: { region: 'gll' },
         invalid: { region: 'foo' },
-        specific: { region: 'de1' },
         without: {}
-      }[regionType];
+      }[region] || { region };
+
+      const scenario = `when called ${{
+        invalid: 'with an invalid',
+        without: 'without a',
+      }[region] || `with "${region}" as the`} signaling region`;
 
-      context(`when called ${regionType === 'without' ? regionType : `with a(n) ${regionType}`} region`, () => {
+      context(scenario, () => {
         let token;
         let cancelablePromise;
+
         beforeEach(() => {
           const identity = randomName();
           token = getToken(identity);
           cancelablePromise = connect(token, Object.assign({}, defaults, regionOptions, { tracks: [] }));
           assert(cancelablePromise instanceof CancelablePromise);
         });
 
-        if (regionType === 'invalid') {
-          it('should return a CancelablePromise that rejects with a SignalingConnectionError', async () => {
-            try {
-              const room = await cancelablePromise;
+        it(`should return a CancelablePromise that ${region === 'invalid' ? 'rejects with a SignalingConnectionError' : 'resolves with a Room'}`, async () => {
+          let error;
+          let room;
```

Comment:
since you are using room in finally block, I suggest explicitly initialize to `null` here